### PR TITLE
Fixed a bug preventing Refresh Tokens from working.

### PR DESCRIPTION
**Pull request by zmoore-usgs**

### CODE REVIEW

File: src/main/java/gov/usgs/wma/mlrauthserver/config/JwtConfig.java [hunk 1]
```diff
@@ -48,13 +49,18 @@ public JwtAccessTokenConverter accessTokenConverter() {
...
 		KeyStoreKeyFactory keyStoreKeyFactory =
 				new KeyStoreKeyFactory(storeFile,this.keystorePassword.toCharArray());
-		converter.setKeyPair(keyStoreKeyFactory.getKeyPair(this.keystoreOAuthKey));
+
+		WaterAuthJwtUserAuthConverter userConverter = new WaterAuthJwtUserAuthConverter();
+		DefaultAccessTokenConverter tokenConverter = new DefaultAccessTokenConverter();
+		tokenConverter.setUserTokenConverter(userConverter);
+
+		JwtAccessTokenConverter jwtConverter = new JwtAccessTokenConverter();
+		jwtConverter.setKeyPair(keyStoreKeyFactory.getKeyPair(this.keystoreOAuthKey));
```
Comment (isuftin) [inline src/main/java/gov/usgs/wma/mlrauthserver/config/JwtConfig.java]: This looks pretty good. Though I wonder if there can be an issue with NPEs or invalid data here..

File: src/main/java/gov/usgs/wma/mlrauthserver/config/WaterAuthJwtUserAuthConverter.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.oauth2.provider.token.DefaultUserAuthenticationConverter;
+
+import static gov.usgs.wma.mlrauthserver.config.WaterAuthJwtEnhancer.EMAIL_JWT_KEY;
+import gov.usgs.wma.mlrauthserver.model.WaterAuthUser;
+
+public class WaterAuthJwtUserAuthConverter extends DefaultUserAuthenticationConverter {	
+	@Override
+	public Authentication extractAuthentication(Map<String, ?> map) {
+		Authentication defaultAuth = super.extractAuthentication(map);
```
Comment (isuftin) [inline src/main/java/gov/usgs/wma/mlrauthserver/config/WaterAuthJwtUserAuthConverter.java]: Check for null on map?

File: src/main/java/gov/usgs/wma/mlrauthserver/config/WaterAuthJwtUserAuthConverter.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+		String email = map.containsKey(EMAIL_JWT_KEY) ? (String)map.get(EMAIL_JWT_KEY) : "";
+		WaterAuthUser user = new WaterAuthUser((String)defaultAuth.getPrincipal(), email, new ArrayList<GrantedAuthority>(defaultAuth.getAuthorities()));
```
Comment (isuftin) [inline src/main/java/gov/usgs/wma/mlrauthserver/config/WaterAuthJwtUserAuthConverter.java]: No issue if email is blank here?

**Comment by zmoore-usgs** [comment]
@isuftin I went back and forth on wrapping the user deserialization stuff within a try-catch but initially left it as-is because I figured if a token did come in without that data it shouldn't be considered valid and should fail. However I think it makes sense to describe this failure better rather than potentially just leaving it as a NPE or something so I think I will add better error checking here.

**Pull request merged**